www/app/locals/local/service.js

```diff
@@ -32,16 +119,26 @@
           'onStateChange': onPlayerStateChange
         }
       });
+      service.nowPlaying = upcomingSongs[actualSong];
     }
 
     function onPlayerReady(event) {
+      eraseSongFromArray($stateParams.localID, upcomingSongs[actualSong].videoID);
+      upcomingSongs.splice(1, 1);
       event.target.playVideo();
     }
 
     function onPlayerStateChange(event) {
       if (event.data == YT.PlayerState.ENDED) {
         actualSong = actualSong + 1;
-        launchNextSong(upcomingSongs[actualSong].id, upcomingSongs[actualSong].title);
+        if (actualSong > upcomingSongs.length-5) {
+          // Queue Ended
```

Comment:
Yeap, some of the next coming up functionality, I will just remove it and add it for another ticket